frontend/src/metabase/dashboard/components/DashboardActions.jsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+
+import Tooltip from "metabase/components/Tooltip";
+import NightModeIcon from "metabase/components/icons/NightModeIcon";
+import FullscreenIcon from "metabase/components/icons/FullscreenIcon";
+import RefreshWidget from "metabase/dashboard/components/RefreshWidget";
+
+export const getDashboardActions = (
+    {
+        isEditing = false,
+        isEmpty = false,
+        isFullscreen,
+        isNightMode,
+        onNightModeChange,
+        onFullscreenChange,
+        refreshPeriod,
+        refreshElapsed,
+        onRefreshPeriodChange
+    }
+) => {
+    const buttons = [];
+
+    if (!isEditing && !isEmpty) {
+        buttons.push(
+            <RefreshWidget
+                data-metabase-event="Dashboard;Refresh Menu Open"
+                className="text-brand-hover"
+                key="refresh"
+                period={refreshPeriod}
+                elapsed={refreshElapsed}
+                onChangePeriod={onRefreshPeriodChange}
+            />
+        );
+    }
+
+    if (!isEditing && isFullscreen) {
+        buttons.push(
+            <Tooltip tooltip={isNightMode ? "Daytime mode" : "Nighttime mode"}>
+                <span
+                    data-metabase-event={"Dashboard;Night Mode;" + !isNightMode}
+                >
+                    <NightModeIcon
+                        className="text-brand-hover cursor-pointer"
+                        key="night"
+                        isNightMode={isNightMode}
+                        onClick={() => onNightModeChange(!isNightMode)}
+                    />
+                </span>
+            </Tooltip>
+        );
+    }
+
+    if (!isEditing && !isEmpty) {
```

Comment:
I mean that there is the same condition already in line 32, so no need to have `!isEditing && !isEmpty` twice. 

Not sure what is the nicest way to avoid `buttons.push`, something like this would work at least but isn't super pretty:
```
return _.flatten([
   condition1 ? [Button1, Button2] : [],
   condition2 ? [Button3] : []
])
```